src/lib/services/instance-stats-service.ts

```diff
@@ -0,0 +1,185 @@
+            | 'environmentStore'
+            | 'strategyStore'
+            | 'contextFieldStore'
+            | 'groupStore'
+            | 'segmentStore'
+            | 'roleStore'
+            | 'settingStore'
+        >,
+        { getLogger }: Pick<IUnleashConfig, 'getLogger'>,
+        versionService: VersionService,
+    ) {
+        this.strategyStore = strategyStore;
+        this.userStore = userStore;
+        this.featureToggleStore = featureToggleStore;
+        this.environmentStore = environmentStore;
+        this.projectStore = projectStore;
+        this.groupStore = groupStore;
+        this.contextFieldStore = contextFieldStore;
+        this.segmentStore = segmentStore;
+        this.roleStore = roleStore;
+        this.versionService = versionService;
+        this.settingStore = settingStore;
+        this.logger = getLogger('services/stats-service.js');
+    }
+
+    async getToggleCount(): Promise<number> {
+        return this.featureToggleStore.count({
+            archived: false,
+        });
+    }
+
+    async hasOIDC(): Promise<boolean> {
+        const settings = await this.settingStore.get(
+            'unleash.enterprise.auth.oidc',
+        );
+
+        return settings?.enabled || false;
+    }
+
+    async hasSAML(): Promise<boolean> {
+        const settings = await this.settingStore.get(
+            'unleash.enterprise.auth.saml',
+        );
+
+        return settings?.enabled || false;
+    }
+
+    async getStats(): Promise<InstanceStats> {
+        const versionInfo = this.versionService.getVersionInfo();
+
+        const [
+            featureToggles,
+            users,
+            projects,
+            contextFields,
+            groups,
+            roles,
+            environments,
+            segments,
+            strategies,
+            SAMLenabled,
+            OIDCenabled,
+        ] = await Promise.all([
+            this.getToggleCount(),
+            this.userStore.count(),
+            this.projectStore.count(),
+            this.contextFieldStore.count(),
+            this.groupStore.count(),
+            this.roleStore.count(),
+            this.environmentStore.count(),
+            this.segmentStore.count(),
+            this.strategyStore.count(),
+            this.hasSAML(),
+            this.hasOIDC(),
+        ]);
+
+        return {
+            timestamp: new Date(),
+            instanceId: versionInfo.instanceId,
+            versionOSS: versionInfo.current.oss,
+            versionEnterprise: versionInfo.current.enterprise,
+            users,
+            featureToggles,
+            projects,
+            contextFields,
+            roles,
+            groups,
+            environments,
+            segments,
+            strategies,
+            SAMLenabled,
+            OIDCenabled,
+        };
+    }
+
+    async getSignedStats(): Promise<InstanceStatsSigned> {
+        const instanceStats = await this.getStats();
+
+        const sum = sha256(
```

Comment:
So I think this is okay for now but how secure do we want to be with this?

When I read `signed`, I think cryptographically secure signature. Do we want to go to those lengths here? This would technically let people read the source and generate these themselves. This interface would let us do that later though

Reply:
I am aware that this is not secure at all. 

The problem is that in case of self-hosted there is little way for us to secure it entirely without doing some kind of network call back to us for a secure signature, but then we could just send the data directly from Unleash. If we base it on anything available locally, one can always trick Unleash to show incorrect numbers, you may even remove all users and re-add them again. 

The thinking I had is that users stupid enough to change the CSV file would not be smart enough to re-create correct hash. Smart users will probably not change this anyway, and if they do, they will be able to trick the numbers regardless.

Reply:
Yeah, that makes sense and I agree with the reasoning here, even with a direct API call there's nothing in theory from stopping someone from just cURLing back whatever data they want.

I think this keeps honest users honest and trying to fight smart, malicious people with root access to Unleash isn't worth it. I'm happy!